src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -687,7 +689,8 @@ export class TerminalInstance implements ITerminalInstance {
 		this._isExiting = true;
 		this._process = null;
 		let exitCodeMessage: string;
-		if (exitCode) {
+
+		if (exitCode && this.configurationService.getValue('terminal.integrated.showExitAlert')) {
```

Comment:
Currently all interactions with the config service is managed within `TerminalConfigHelper`, could you change to use `this._configHelper.config.showExitAlert` instead? You'll also need to add it to the interface defined in terminal.ts